### PR TITLE
Fix #10222: Move scala.quoted.internal to scala.quoted.internal

### LINKED ISSUE #10222: Move scala.internal.quoted to scala.quoted.internal?
If we could avoid adding even more stuff inside scala that would be great, and there's nothing else in scala.internal, wdyt @nicolasstucki ?

### PR DESCRIPTION
**Pull request by nicolasstucki**

### CODE REVIEW

File: compiler/src/dotty/tools/dotc/transform/Splicer.scala [hunk 1]
```diff
@@ -323,10 +323,10 @@ object Splicer {
     }
 
     private def interpretQuote(tree: Tree)(implicit env: Env): Object =
-      new scala.quoted.internal.Expr(Inlined(EmptyTree, Nil, QuoteUtils.changeOwnerOfTree(tree, ctx.owner)).withSpan(tree.span), QuoteContextImpl.scopeId)
+      new dotty.tools.dotc.quoted.ExprImpl(Inlined(EmptyTree, Nil, QuoteUtils.changeOwnerOfTree(tree, ctx.owner)).withSpan(tree.span), QuoteContextImpl.scopeId)
```
Comment (liufengyun) [inline compiler/src/dotty/tools/dotc/transform/Splicer.scala:326]:
```suggestion
      new quoted.ExprImpl(Inlined(EmptyTree, Nil, QuoteUtils.changeOwnerOfTree(tree, ctx.owner)).withSpan(tree.span), QuoteContextImpl.scopeId)
```

File: compiler/src/dotty/tools/dotc/transform/Splicer.scala [hunk 2]
```diff
@@ -323,10 +323,10 @@ object Splicer {
...
 
     private def interpretTypeQuote(tree: Tree)(implicit env: Env): Object =
-      new scala.quoted.internal.Type(QuoteUtils.changeOwnerOfTree(tree, ctx.owner), QuoteContextImpl.scopeId)
+      new dotty.tools.dotc.quoted.TypeImpl(QuoteUtils.changeOwnerOfTree(tree, ctx.owner), QuoteContextImpl.scopeId)
```
Comment (liufengyun) [inline compiler/src/dotty/tools/dotc/transform/Splicer.scala:329]:
```suggestion
      new quoted.TypeImpl(QuoteUtils.changeOwnerOfTree(tree, ctx.owner), QuoteContextImpl.scopeId)
```

File: compiler/src/dotty/tools/dotc/quoted/QuoteContextImpl.scala [hunk 3]
```diff
@@ -71,7 +72,7 @@ class QuoteContextImpl private (ctx: Context) extends QuoteContext, scala.intern
             case _ => false
         def asExpr: scala.quoted.Expr[Any] =
           if self.isExpr then
-            new scala.quoted.internal.Expr(self, QuoteContextImpl.this.hashCode)
+            new dotty.tools.dotc.quoted.ExprImpl(self, QuoteContextImpl.this.hashCode)
```
Comment (liufengyun) [inline compiler/src/dotty/tools/dotc/quoted/QuoteContextImpl.scala:75]:
```suggestion
            new ExprImpl(self, QuoteContextImpl.this.hashCode)
```

File: compiler/src/dotty/tools/dotc/quoted/QuoteContextImpl.scala [hunk 4]
```diff
@@ -315,11 +316,11 @@ class QuoteContextImpl private (ctx: Context) extends QuoteContext, scala.intern
     object TermMethodsImpl extends TermMethods:
       extension (self: Term):
         def seal: scala.quoted.Expr[Any] =
-          if self.isExpr then new scala.quoted.internal.Expr(self, QuoteContextImpl.this.hashCode)
+          if self.isExpr then new dotty.tools.dotc.quoted.ExprImpl(self, QuoteContextImpl.this.hashCode)
```
Comment (liufengyun) [inline compiler/src/dotty/tools/dotc/quoted/QuoteContextImpl.scala:319]:
```suggestion
          if self.isExpr then new ExprImpl(self, QuoteContextImpl.this.hashCode)
```

File: compiler/src/dotty/tools/dotc/quoted/QuoteContextImpl.scala [hunk 5]
```diff
@@ -315,11 +316,11 @@ class QuoteContextImpl private (ctx: Context) extends QuoteContext, scala.intern
...
           else throw new Exception("Cannot seal a partially applied Term. Try eta-expanding the term first.")
 
         def sealOpt: Option[scala.quoted.Expr[Any]] =
-          if self.isExpr then Some(new scala.quoted.internal.Expr(self, QuoteContextImpl.this.hashCode))
+          if self.isExpr then Some(new dotty.tools.dotc.quoted.ExprImpl(self, QuoteContextImpl.this.hashCode))
```
Comment (liufengyun) [inline compiler/src/dotty/tools/dotc/quoted/QuoteContextImpl.scala:323]:
```suggestion
          if self.isExpr then Some(new ExprImpl(self, QuoteContextImpl.this.hashCode))
```

File: compiler/src/dotty/tools/dotc/quoted/QuoteContextImpl.scala [hunk 6]
```diff
@@ -1002,7 +1003,7 @@ class QuoteContextImpl private (ctx: Context) extends QuoteContext, scala.intern
 
     object TypeTree extends TypeTreeModule:
       def of[T <: AnyKind](using tp: scala.quoted.Type[T]): TypeTree =
-        tp.asInstanceOf[scala.quoted.internal.Type].typeTree
+        tp.asInstanceOf[dotty.tools.dotc.quoted.TypeImpl].typeTree
```
Comment (liufengyun) [inline compiler/src/dotty/tools/dotc/quoted/QuoteContextImpl.scala:1006]:
```suggestion
        tp.asInstanceOf[TypeImpl].typeTree
```

File: compiler/src/dotty/tools/dotc/quoted/QuoteContextImpl.scala [hunk 7]
```diff
@@ -1571,7 +1572,7 @@ class QuoteContextImpl private (ctx: Context) extends QuoteContext, scala.intern
 
     object TypeRepr extends TypeReprModule:
       def of[T <: AnyKind](using tp: scala.quoted.Type[T]): TypeRepr =
-        tp.asInstanceOf[scala.quoted.internal.Type].typeTree.tpe
+        tp.asInstanceOf[dotty.tools.dotc.quoted.TypeImpl].typeTree.tpe
```
Comment (liufengyun) [inline compiler/src/dotty/tools/dotc/quoted/QuoteContextImpl.scala:1575]:
```suggestion
        tp.asInstanceOf[TypeImpl].typeTree.tpe
```

File: compiler/src/dotty/tools/dotc/quoted/QuoteContextImpl.scala [hunk 8]
```diff
@@ -1608,7 +1609,7 @@ class QuoteContextImpl private (ctx: Context) extends QuoteContext, scala.intern
         def seal: scala.quoted.Type[_] = self.asType
 
         def asType: scala.quoted.Type[?] =
-          new scala.quoted.internal.Type(Inferred(self), QuoteContextImpl.this.hashCode)
+          new dotty.tools.dotc.quoted.TypeImpl(Inferred(self), QuoteContextImpl.this.hashCode)
```
Comment (liufengyun) [inline compiler/src/dotty/tools/dotc/quoted/QuoteContextImpl.scala:1612]:
```suggestion
          new TypeImpl(Inferred(self), QuoteContextImpl.this.hashCode)
```

File: compiler/src/dotty/tools/dotc/quoted/QuoteContextImpl.scala [hunk 9]
```diff
@@ -2623,11 +2624,11 @@ class QuoteContextImpl private (ctx: Context) extends QuoteContext, scala.intern
 
   def unpickleExpr[T](pickled: String | List[String], typeHole: (Int, Seq[Any]) => scala.quoted.Type[?], termHole: (Int, Seq[Any], scala.quoted.QuoteContext) => scala.quoted.Expr[?]): scala.quoted.Expr[T] =
     val tree = PickledQuotes.unpickleTerm(pickled, typeHole, termHole)(using reflect.rootContext)
-    new scala.quoted.internal.Expr(tree, hash).asInstanceOf[scala.quoted.Expr[T]]
+    new dotty.tools.dotc.quoted.ExprImpl(tree, hash).asInstanceOf[scala.quoted.Expr[T]]
```
Comment (liufengyun) [inline compiler/src/dotty/tools/dotc/quoted/QuoteContextImpl.scala:2627]:
```suggestion
    new ExprImpl(tree, hash).asInstanceOf[scala.quoted.Expr[T]]
```

File: compiler/src/dotty/tools/dotc/quoted/QuoteContextImpl.scala [hunk 10]
```diff
@@ -2623,11 +2624,11 @@ class QuoteContextImpl private (ctx: Context) extends QuoteContext, scala.intern
...
 
   def unpickleType[T <: AnyKind](pickled: String | List[String], typeHole: (Int, Seq[Any]) => scala.quoted.Type[?], termHole: (Int, Seq[Any], scala.quoted.QuoteContext) => scala.quoted.Expr[?]): scala.quoted.Type[T] =
     val tree = PickledQuotes.unpickleTypeTree(pickled, typeHole, termHole)(using reflect.rootContext)
-    new scala.quoted.internal.Type(tree, hash).asInstanceOf[scala.quoted.Type[T]]
+    new dotty.tools.dotc.quoted.TypeImpl(tree, hash).asInstanceOf[scala.quoted.Type[T]]
```
Comment (liufengyun) [inline compiler/src/dotty/tools/dotc/quoted/QuoteContextImpl.scala:2631]:
```suggestion
    new TypeImpl(tree, hash).asInstanceOf[scala.quoted.Type[T]]
```

**Review by liufengyun — APPROVED** [review]
LGTM

**Comment by nicolasstucki** [comment]
I'll clean up those paths later. I may still change something in the compiler implementation.

**Pull request merged**